### DISCUSSION TITLE
Constructing climographs

### DISCUSSION THREAD

**Discussion by rdstern** [IDEMSInternational/R-Instat#8310] [Climatic]
A climograph is a simple climatic product.  I give examples below.  I started, naively, thinking that we should be able to construct them using our general bar chart/line chart capabilities with R-Instat.  I would still like to be able to do this, but am failing miserably.  I now think we need a special dialogue as we also gradually build up the general graphics capabilities.

First, some examples - from Wikipedia - of course, where it says:

`Climographs display data for two variables: (a) monthly average temperature and (b) monthly average precipitation. These are useful tools to quickly describe a location's climate."

![image](https://user-images.githubusercontent.com/11226469/236766493-413b759e-cd74-452c-a65e-baef618e6c32.png)

This is a basic one, with just one temperature line.  But you could also have avarage max and min.  So, still from Wikipedia:

![image](https://user-images.githubusercontent.com/11226469/236768174-69c6d8bd-792e-4caf-b25c-b25eabc58c48.png)

And this [website](https://grayepc.weebly.com/uploads/2/3/6/2/23621820/climate_-_day_3.pdf) gives information that can easily be translated into exercises.

In the early days of ggplot, the author was against permitting graphs where the left and right axes were of unrelated variables.  But this is possible - indeed quite straightforward now and there are good stack-overflow questions on the topic which give a range of examples, for example [here](https://stackoverflow.com/questions/27505156/climatograph-with-r-and-ggplot) and [here](https://stackoverflow.com/questions/56259187/r-climograph-with-ggplot2-change-values-of-precipitaion-to-fit-second-axis)  and [here](https://stackoverflow.com/questions/73181349/create-climate-diagram-in-r-with-temperature-and-precipitation) and [here](https://stackoverflow.com/questions/56701265/plotting-two-seperate-barcharts-in-one-graph-using-ggplot2)

One variation, introduced here is for multiple years:

![image](https://user-images.githubusercontent.com/11226469/236775503-be0a08a3-bb92-4cee-b9c9-8f03b53932f5.png)

On names I have seen climatoigraph, climograph, climatogram as well as climate graph.

Once we have a dialogue we may wish to extend it to add Walter-Leith diagrams (another button at the top), for which there is now a ggplot2 function in the relatively new climaemet package.  This updates the function in climatol.

Here is possible data for a dialogue:
[moz3a.zip](https://github.com/africanmathsinitiative/R-Instat/files/11419883/moz3a.zip)

It could be monthly data for each year.  Usually it will be average monthly data, so just 12 rows for each station.  If monthly data, then I suggest the year and month be made into a Date variable, so the x-axis can be displayed nicely - as shown in one example.

I suggest it should be easy to construct the dialogue, because we have similar controls elsewhere.  So:

a) Usual data selector.
b) Single receiver with label `Precip:`
c) The Temperature receivers depend on whether we are using tmax and tmin or tmean.  I think we have that choice also in the Climatic > Prepare > Transform > Degrees dialogue:

![image](https://user-images.githubusercontent.com/11226469/236790466-74c86565-9038-4404-aceb-316f00feda33.png)

d) Then another single receiver titled `Month/Date: `

e) Finally - so far for the right-hand side - a copy of the `Facet By (Optional)` Control, from the PICSA graphs dialogue.

f) On the lhs there will be some options I am sure.  Ideally we will be producing a ggplot graph, so let's hope the Plot Options could work?

Now while perhaps @Vitalis95 produces the dialogue @COLIEWO could examine the R-code needed for some examples?

I suggest the menu item and dialogue title can be called Climograph.  And it can gi in the Climatic > Describe menu below Temperature.  That meny needs a lot of improvement but let's do that later.

**Comment by COLIEWO** [comment]
I have looked at the reference materials shared and so far the challenge is getting the line plot for temperature to take the secondary axis values so its more visible.
I used the data shared and combined the bar and line graphs as below with the secondary axis being a multiple of the primary axis for precipitation.
`data_R_instat_by_Station_Name_year_month_abbr_by_Station_Name_month_abbr <- data_book$get_data_frame(data_name="data_R_instat_by_Station_Name_year_month_abbr_by_Station_Name_month_abbr")
last_graph <- ggplot2::ggplot(data_R_instat_by_Station_Name_year_month_abbr_by_Station_Name_month_abbr) + ggplot2::geom_bar(aes(x=month_abbr, y=precip, fill=Station_Name),position="dodge", stat="identity") + ggplot2::geom_line(aes(x=month_abbr, y=tmax, colour=Station_Name, group=Station_Name)) + ggplot2::scale_y_continuous(sec.axis = sec_axis(~./5,name = 'Temp')) +  theme_grey()
data_book$add_object(data_name="data_R_instat_by_Station_Name_year_month_abbr_by_Station_Name_month_abbr", object_name="last_graph", object_type_label="graph", object_format="image", object=check_graph(graph_object=last_graph))
data_book$get_object_data(data_name="data_R_instat_by_Station_Name_year_month_abbr_by_Station_Name_month_abbr", object_name="last_graph", as_file=TRUE)
rm(list=c("last_graph", "data_R_instat_by_Station_Name_year_month_abbr_by_Station_Name_month_abbr"))`
![clim1](https://github.com/africanmathsinitiative/R-Instat/assets/124562645/92e70098-9e1b-4fe3-8a42-795530ae2335)

**Comment by COLIEWO** [comment]
The solution to the line graph taking values from the secondary axis would be to transform the values by the same factor that transformed the axis, in my case `ggplot2::geom_line(aes(x=month_abbr, y=tmax*5, colour=Station_Name, group=Station_Name))` since we had divided by 5 to get the secondary axis.
![clim2](https://github.com/africanmathsinitiative/R-Instat/assets/124562645/154016f9-26c8-435c-baaa-caa228c97f51)

**Comment by rdstern** [comment]
@jkmusyoka this is the discussion that I hope can now be extended and developed in a more modern way as we discussed.  And some options are above.  I hope it can become part of an effectively new menu in Climatic > Describe.  That's described in a separate discussion.
